pennylane/templates/embeddings/iqp.py

```diff
@@ -0,0 +1,249 @@
+
+        **Using a custom entangler pattern**
+
+        A custom entangler pattern can be used by specifying the ``pattern`` argument. A pattern has to be
+        a nested list of dimension ``(K, 2)``, where ``K`` is the number of entanglers to apply.
+
+        .. code-block:: python
+
+            pattern = [[1, 2], [0, 2], [1, 0]]
+
+            @qml.qnode(dev)
+            def circuit(features=None):
+                IQPEmbedding(features=features, wires=range(3), pattern=pattern)
+                return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            circuit(features=[1., 2., 3.])
+
+        Since diagonal gates commute, the order of the wire pairs has no effect on the result.
+
+        .. code-block:: python
+
+            from pennylane import numpy as np
+
+            pattern1 = [[1, 2], [0, 2], [1, 0]]
+            pattern2 = [[1, 0], [0, 2], [1, 2]]
+
+            @qml.qnode(dev)
+            def circuit(features=None, pattern=None):
+                IQPEmbedding(features=features, wires=range(3), pattern=pattern, n_repeats=3)
+                return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            res1 = circuit(features=[1., 2., 3.], pattern=pattern1)
+            res2 = circuit(features=[1., 2., 3.], pattern=pattern2)
+
+            assert np.allclose(res1, res2)
+
+        **Repeating the embedding**
+
+        The embedding can be repeated by specifying the ``n_repeats`` argument:
+
+        .. code-block:: python
+
+            @qml.qnode(dev)
+            def circuit(features=None):
+                IQPEmbedding(features=features, wires=range(3), n_repeats=4)
+                return [qml.expval(qml.PauliZ(w)) for w in range(3)]
+
+            circuit(features=[1., 2., 3.])
+
+        Every repetition uses exactly the same quantum circuit.
+
+    """
+    #############
+    # Input checks
+
+    wires = check_wires(wires)
+
+    check_no_variable(features, msg="'features' cannot be differentiable")
+
+    expected_shape = (len(wires),)
+    check_shape(
+        features,
+        expected_shape,
+        msg="'features' must be of shape {}; got {}" "".format(expected_shape, get_shape(features)),
+    )
+
+    check_type(
+        n_repeats, [int], msg="'n_repeats' must be an integer; got type {}".format(type(n_repeats))
+    )
+
+    check_type(
+        pattern,
+        [Iterable, type(None)],
+        msg="'pattern' must be None or a list of wire pairs; got {}".format(pattern),
+    )
+
+    if pattern is not None:
+
+        # check type
+        for p in pattern:
+            check_type(
+                p,
+                [Iterable],
+                msg="'pattern' must be None or a list of wire pairs; got {}".format(pattern),
+            )
+
+            for w in p:
+                check_type(
+                    w,
+                    [int],
+                    msg="'pattern' must be None or a list of wire pairs; got {}".format(pattern),
+                )
```

Comment:
Yes, and I would like to keep it that way: We would have to convert to list otherwise. If I pass a numpy array, the integers are numpy integers, and we had trouble with the qiskit device. 

This will be resolved by the upcoming upgrade of the wire management.